sklearn/cluster/k_means_.py

```diff
@@ -98,7 +97,7 @@ def _k_init(X, n_clusters, x_squared_norms, random_state, n_local_trials=None):
     # Initialize list of closest distances and calculate current potential
     closest_dist_sq = euclidean_distances(
         centers[0, np.newaxis], X, Y_norm_squared=x_squared_norms,
-        squared=True)
```

Comment:
I think the best choice would be to add a check_input flag to `_k_init`. Then if this flag is true the input is checked at the start of the _k_init function. For minibatch this flag can be set to false and for regular kmeans this flag can be set to true (or we can just put the check in kmeans as well). In all cases check_input would be set to false in the euclidian_distance calls inside _k_init.